lib/ansible/modules/cloud/amazon/dynamodb_table.py

```diff
@@ -293,13 +294,23 @@ def delete_dynamo_table(connection, module):
         module.exit_json(**result)
 
 
+def _is_table_not_found_error(error):
+    msg = error.message
```

Comment:
I don't think python3 has `error.message` 

We should probably use `error.response['Error']['Code']`

Reply:
Weird, I tested it manually using python 3.6.
I'll add placebo test for this function to double check.

Reply:
@willthames added test to check that boto throws `boto.exception.JSONResponseError` exception.

Reply:
`dynamo_table_exists` function should've checked `error_code` instead of message:
```python
def dynamo_table_exists(table):
    try:
        table.describe()
        return True

    except JSONResponseError as e:
        if e.error_code == 'ResourceNotFoundException':
            return False
        else:
            raise e

```

Not sure why @loia did check on message. DynamoDB [specification](https://github.com/boto/botocore/blob/cccfdf86bc64877ad41e0af74b752b8a49fc4d33/botocore/data/dynamodb/2012-08-10/service-2.json#L200-L213) clearly sates that `DescribeTable` can only either throw Internal error or `ResourceNotFoundException`, which should've been checked instead. Though to preserve backward compatibility and make this change easier to merge, I left this function as is and just added 1 additional edge case, which is only reproducible   on dynamodb local instance.

@willthames please let me know if you would prefer to use `error_code` as it's shown above.